taxonomy-editor-frontend/src/pages/startproject/index.tsx

```diff
@@ -1,62 +1,62 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
 import {
   Typography,
   Box,
   Grid,
   TextField,
-  Stack,
-  Autocomplete,
   Snackbar,
   Alert,
   CircularProgress,
   Button,
+  InputLabel,
+  FormControl,
+  Select,
+  MenuItem,
 } from "@mui/material";
-import { useState } from "react";
-import { useNavigate } from "react-router-dom";
+
 import { TAXONOMY_NAMES } from "../../constants";
 import { createBaseURL } from "../editentry/createURL";
-import { toSnakeCase } from "../../components/interConvertNames";
+import { toSnakeCase } from "../../utils";
 
 const StartProject = () => {
   const [branchName, setBranchName] = useState("");
-  const [taxonomyName, setTaxonomyName] = useState(null);
+  const [taxonomyName, setTaxonomyName] = useState("");
   const [description, setDescription] = useState("");
   const [loading, setLoading] = useState(false);
-  const [errorMessage, setErrorMessage] = useState(null);
+  const [errorMessage, setErrorMessage] = useState("");
   const navigate = useNavigate();
 
   const handleSubmit = () => {
-    const baseUrl = createBaseURL(taxonomyName, branchName);
+    if (!taxonomyName || !branchName) return;
+
+    const baseUrl = createBaseURL(toSnakeCase(taxonomyName), branchName);
     setLoading(true);
     const dataToBeSent = { description: description };
-    fetch(baseUrl + "import", {
+
+    fetch(`${baseUrl}import`, {
       method: "POST",
       headers: { "Content-Type": "application/json" },
       body: JSON.stringify(dataToBeSent),
     })
       .then(async (response) => {
         const responseBody = await response.json();
-        if (!response.ok && responseBody.detail) {
-          throw new Error(responseBody.detail);
+        if (!response.ok) {
+          throw new Error(responseBody?.detail ?? "Unable to import");
```

Comment:
yeah, so, `!response.ok` could happen for a number of reasons we may not control, thus the `responseBody` is not guaranteed to exist so all this line is doing is adding a default value in case the `responseBody?.detail` doesn't exist